homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,178 @@
+    API_STATION_LOCATION,
+    API_STATION_MAC_ADDRESS,
+    API_STATION_NAME,
+    API_STATION_TYPE,
+    DOMAIN,
+    ENTITY_MAC_ADDRESS,
+    ENTITY_STATION_NAME,
+)
+
+CONFIG_USER = "user"
+
+CONFIG_LOCATION = "location"
+CONFIG_LOCATION_LATITUDE = "latitude"
+CONFIG_LOCATION_LONGITUDE = "longitude"
+CONFIG_LOCATION_RADIUS = "radius"
+CONFIG_LOCATION_RADIUS_DEFAULT = DistanceConverter.convert(
+    0.5,
+    UnitOfLength.MILES,
+    UnitOfLength.METERS,
+)
+
+CONFIG_STATION = "station"
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for the Ambient Weather Network integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Construct the config flow."""
+
+        self._longitude = 0.0
+        self._latitude = 0.0
+        self._radius = 0.0
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step to select the location."""
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._latitude = user_input[CONFIG_LOCATION][CONFIG_LOCATION_LATITUDE]
+            self._longitude = user_input[CONFIG_LOCATION][CONFIG_LOCATION_LONGITUDE]
+            self._radius = DistanceConverter.convert(
+                user_input[CONFIG_LOCATION][CONFIG_LOCATION_RADIUS],
+                UnitOfLength.METERS,
+                UnitOfLength.MILES,
+            )
+            return await self.async_step_station()
+
+        schema: vol.Schema = self.add_suggested_values_to_schema(
+            vol.Schema(
+                {
+                    vol.Required(
+                        CONFIG_LOCATION,
+                    ): LocationSelector(LocationSelectorConfig(radius=True)),
+                }
+            ),
+            {
+                CONFIG_LOCATION: {
+                    CONFIG_LOCATION_LATITUDE: self.hass.config.latitude,
+                    CONFIG_LOCATION_LONGITUDE: self.hass.config.longitude,
+                    CONFIG_LOCATION_RADIUS: CONFIG_LOCATION_RADIUS_DEFAULT,
+                }
+            },
+        )
+
+        return self.async_show_form(
+            step_id=CONFIG_USER,
+            data_schema=schema,
+            errors=errors,
+        )
+
+    @staticmethod
+    def get_station_name(station: dict[str, Any]) -> str:
+        """Pick a station name.
+
+        Station names can be empty, in which case we construct the name from
+        the location and device type.
+        """
+        if name := station.get(API_STATION_INFO, {}).get(API_STATION_NAME):
+            return str(name)
+        location = (
+            station.get(API_STATION_INFO, {})
+            .get(API_STATION_COORDS, {})
+            .get(API_STATION_LOCATION)
+        )
+        station_type = station.get(API_LAST_DATA, {}).get(API_STATION_TYPE)
+        return f"{location}{'' if location is None or station_type is None else ' '}{station_type}"
```

Comment:
This function can be outside of this object. Can the name of a station also change?

Reply:
Moved.